source/fe/fe_enriched.cc

```diff
@@ -1048,6 +1049,451 @@ FE_Enriched<dim, spacedim>::InternalData::get_fe_output_object(
+                {
+                  if (fe_sets[j] == color_list)
+                    {
+                      found = true;
+                      cell->set_active_fe_index(j);
+                      break;
+                    }
+                }
+
+              if (!found)
+                {
+                  fe_sets.push_back(color_list);
+                  cell->set_active_fe_index(fe_sets.size() - 1);
+                }
+            }
+          ++map_index; // map_index should be unique to cells
+        }
+
+
+      /*
+       * Treat interface between enriched cells specially,
+       * until #1496 (https://github.com/dealii/dealii/issues/1496) is resolved.
+       * Each time we build constraints at the
+       * interface between two different FE_Enriched, we look for the least
+       * dominating FE via
+       * hp::FECollection< dim, spacedim >::find_least_face_dominating_fe().
+       * If we don't take further actions, we may find a dominating FE that
+       * is too restrictive, i.e. enriched FE consisting of only FE_Nothing.
+       * New elements needs to be added to FECollection object to help
+       * find the correct enriched FE element underlying the spaces in the
+       * adjacent cells. This is done by creating an appropriate
+       * set in fe_sets and a call to the function
+       * make_fe_collection_from_colored_enrichments at a later stage.
+       *
+       * Consider a domain with three predicates and hence with three different
+       * enrichment functions. Let the enriched finite element of a cell with
+       * enrichment functions 1 and 2 be represented by [1 1 0], with the last
+       * entry as zero since the 3rd enrichment function is not relevant for
+       * the cell. If the interface has enriched FE [1 0 1] and [0 1 1]
+       * on adjacent cells, an enriched FE [0 0 1] should exist and is
+       * found as the dominating finite element for the two cells by
+       * DoFTools::make_hanging_node_constraints using a call to the function
+       * hp::FECollection::find_least_face_dominating_fe.
+       * Denoting the fe set in adjacent cells as {1,3} and {2,3}, this
+       * implies that an fe set {3} needs to be added! Based on the
+       * predicate configuration, this may not be automatically done without
+       * the following special treatment.
+       */
+
+      // loop through faces
+      for (auto cell = dof_handler.begin_active(); cell != dof_handler.end();
+           ++cell)
+        for (unsigned int face = 0; face < GeometryInfo<dim>::faces_per_cell;
+             ++face)
+          {
+            unsigned int fe_index = cell->active_fe_index();
+            auto         fe_set   = fe_sets[fe_index];
+
+            // find active indices of neighboring cells
+            if (!cell->at_boundary(face))
+              {
+                unsigned int nbr_fe_index =
+                  cell->neighbor(face)->active_fe_index();
```

Comment:
before doing his, you should visit each face only once, to that end you can, for example, use material id which you set above, i.e.
```
if (cell->set_material_id < nbr_cell->material_id)
   ... do something
```